worksheet_D/oxo.py

```diff
@@ -0,0 +1,108 @@
+
+class OxoBoard:
+    def __init__(self):
+        #Initialises board array
+        self.gameBoard = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+
+
+    def get_square(self, x, y):
+        #Returns 0, 1 or 2 depending on the contents of the specified square.
+        return  self.gameBoard[x][y]
+
+
+    def set_square(self, x, y, mark):
+        #Checks content of square and fills empty spots with marks and returns True, is space is taken returns False
+        if self.gameBoard[x][y] == 0:
+            self.gameBoard[x][y] = mark
+            return True
+        else:
+            return False
+
+
+    def is_board_full(self):
+        #Checks every square to see if there are any empty squares left, returns True if none are empty
+        for x in xrange(3):
+            for y in xrange(3):
+                if self.gameBoard[x][y] == 0:
+                    return False
+        return True
```

Comment:
May be worth putting a comment here: if this point is reached, there are no empty squares on the board